components/infobox/wikis/smite/infobox_unit_god_custom.lua

```diff
@@ -0,0 +1,200 @@
+---
+-- @Liquipedia
+-- wiki=smite
+-- page=Module:Infobox/Unit/Champion
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local GodWL = require('Module:GodWL')
+local Lua = require('Module:Lua')
+local Math = require('Module:MathUtil')
+local Namespace = require('Module:Namespace')
+local String = require('Module:StringUtils')
+local Template = require('Module:Template')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector')
+local Unit = Lua.import('Module:Infobox/Unit')
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Breakdown = Widgets.Breakdown
+local Cell = Widgets.Cell
+local Center = Widgets.Center
+local Header = Widgets.Header
+local Title = Widgets.Title
+
+---@class SmiteUnitInfobox: UnitInfobox
+local CustomGod = Class.new(Unit)
+
+local CustomInjector = Class.new(Injector)
+
+local FAVOR_ICON = '[[File:Smite Currency Favor.png|x20px|Favor|link=Favor]]'
+local GEMS_ICON = '[[File:Smite Currency Gems.png|x20px|Gems|link=Gems]]'
+
+---@param frame Frame
+---@return Html
+function CustomGod.run(frame)
+	local unit = CustomGod(frame)
+	unit:setWidgetInjector(CustomInjector(unit))
+	unit.args.informationType = 'God'
+
+	return unit:createInfobox()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	local args = self.caller.args
+	if id == 'header' then
+		return {
+			Header {
+				name = args.godname,
+				subHeader = args.title,
+				image = args.image,
+				imageDefault = args.default,
+				imageDark = args.imagedark or args.imagedarkmode,
+				imageDefaultDark = args.defaultdark or args.defaultdarkmode,
+			},
+		}
+	elseif id == 'caption' then
+		table.insert(widgets, Center {content = {args.quote}})
+	elseif id == 'type' then
+		local breakDownContents = {}
+		local pantheon = args.pantheon
+		if not String.isEmpty(pantheon) then
+			pantheon = '<b>Pantheon</b><br>' ..
+				Template.safeExpand(mw.getCurrentFrame(), 'Pantheon icon', {pantheon}, '')
+			table.insert(breakDownContents, pantheon)
+		end
+		local class = args.class
+		if not String.isEmpty(class) then
+			class = '<b>Class</b><br>' ..
+				Template.safeExpand(mw.getCurrentFrame(), 'Class icon', {class},
+					'')
+			table.insert(breakDownContents, class)
+		end
+		local powerType = args.powertype
+		if not String.isEmpty(powerType) then
+			powerType = '<b>Power Type</b><br>' ..
+				Template.safeExpand(mw.getCurrentFrame(), 'Power icon', {powerType}, '')
+			table.insert(breakDownContents, powerType)
+		end
```

Comment:
- please no template expand on a new wiki ...
- this is basically 3 times the same, make it a local function to reduce code duplication, something like the below

```suggestion
		local toBreakDownCell = function(key, title, fn)
			if String.isEmpty(args[key]) then return end
			return '<b>' .. title .. '</b><br>' .. fn(args[key])
		end

		local breakDownContents = Array.append({},
			toBreakDownCell('pantheon', 'Pantheon', PantheonIconModule.run)
			toBreakDownCell('class', 'Class', ClassModule.run)
			toBreakDownCell('powertype', 'Power Type', PowerIconModule.run)
		)
```
possibly malke the different icon modules one???
idk what the crap templates do